src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/main.js

```diff
@@ -1150,10 +1160,37 @@ define([
                 };
             }
         },
+
         destroy: function() {
             if (!!this.preview) {
                 Preview.destroy(this.preview);
             }
+        },
+
+        tabChangeAllowed: function () {
+            var allowed = $.Deferred();
+            if (!this.saved) {
+                // show warning dialog
+                this.sandbox.emit('sulu.overlay.show-warning',
+                    'sulu.overlay.be-careful',
+                    'content.template.dialog.content',
+
+                    function() {
+                        // cancel callback
+                        allowed.fail();
```

Comment:
And if we are using Deferreds I would say we also use the `resolve` method here, and distinguish the "ok"  and "cancel" buttons from the overlay by passing different options. `fail` doesn't feel correct, because it's not really an error if somebody clicks cancel.